cpu/stm32/periph/adc_f3.c

```diff
@@ -134,13 +134,7 @@ int adc_init(adc_t line)
     if (!(dev(line)->CR & ADC_CR_ADEN)) {
         /* Enable ADC internal voltage regulator and wait for startup period */
         dev(line)->CR |= ADC_CR_ADVREGEN;
-#if IS_USED(MODULE_ZTIMER_USEC)
-        ztimer_sleep(ZTIMER_USEC, ADC_T_ADCVREG_STUP_US);
-#else
-        /* to avoid using ZTIMER_USEC unless already included round up the
-           internal voltage regulator start up to 1ms */
-        ztimer_sleep(ZTIMER_MSEC, 1);
-#endif
+        ztimer_sleep_usecs(ADC_T_ADCVREG_STUP_US);
```

Comment:
Note that with the exception of the possibility to degrade to second resolution, this is already doing what the convenience functions do. So the question on whether automatic clock selection is acceptable needs to be answered independent of this PR.

There are a lot of drivers that call into ztimer exactly once for a sleep after a reset, and the drivers don't care at all if the delay gets rounded up to milliseconds, or even seconds. (But I agree that for user experience, rounding up to seconds might indeed be a footgun.) For low power use cases it could be beneficial to allow applications to run without pulling in `ztimer_usec` at all. And on the other hand, a user with a low-on-memory device might care not about the power consumption benefit of adding `ztimer_msec`, but care a lot about the additional memory this would use.

So I think there are use cases for this. So why not provide a convenience function for this to make it easier to cater these use cases.

Reply:
I dropped the degrading to seconds. It is now identical to the code before, just easier to read. I also added some more visible warnings.

Reply:
I see the use case. I still think that we cannot have a "ztimer_sleep_usecs()" that behaves as specified. (the name alone would make users *use* it, and probably be surprised.)

Can we boil this down to _just_ `ztimer_sleep_msec()` that just falls back to use ZTIMER_USEC if MSEC is not available?
If users don't care about the accuracy, that should be fine.
And, multiplication is faster than division. :)

Having an API that is called "sleep for usecs" but sometimes it actually sleeps quantized to an msec timer is weird.

Reply:
> And, multiplication is faster than division. 

For the use case it has this shouldn't matter, there should be a `DRIVER_FOO_RESET_SLEEP` defined somewhere, so that the compiler does constant folding anyway.

> the name alone would make users use it, and probably be surprised

Maybe we should just find a better name. `ztimer_sleep_usec_with_footgun()` or `ztimer_sleep_usec_unprecise()`?

Reply:
`ztimer_sleep_at_least()` because when initializing drivers, I don't care how long the sleep takes as long as it sleeps for at least the required amount of time.

Reply:
Note that also `ztimer_sleep()` has at-least-semantics, as there is little ztimer could do for low prio tasks in case higher priority tasks would kick in when the sleep is over. The difference is only that this sleep can take longer even in absence of load.

I fear if the is a special at-least function, users start assuming the regular version has no at-least-semantics.

Reply:
ztimer_sleep() has at-least semantic but also best effort (if there isn't a higher prio task and the sleep isn't to short to manage it should be within 0-2 units)

Reply:
I added a `_relaxed` postfix to all functions, as for each of them at least one assumption has to be relaxed regarding the behavior.